### PR TITLE
Add dashboard next into RHChe packaging

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
This PR adds a dashboard next to RHChe packaging.
Note that Dashboard next is going to have a different customization mechanism, but until it's not implemented - we're going to show default values from upstream on Dashboard Next. Customization should be a requirement before we start Dashboard next promoting officially on che.openshift.io.

### What issues does this PR fix or reference?
Upstream changes were done in https://github.com/eclipse/che/pull/17041

### How have you tested this PR?
I've tested docker build with Dockerfile:

<details>

<summary>Build logs</summary>

```
~/projects/rh-che/dockerfiles/che-fabric8 dashboardNext > docker build -t sleshche:rh-che-local --file=Dockerfile .                                                14:03:48
Sending build context to Docker daemon  78.16MB
Step 1/27 : ARG CHE_DASHBOARD_VERSION=next
Step 2/27 : ARG CHE_DASHBOARD_NEXT_VERSION=next
Step 3/27 : ARG CHE_WORKSPACE_LOADER_VERSION=next
Step 4/27 : FROM quay.io/eclipse/che-dashboard:${CHE_DASHBOARD_VERSION} as che_dashboard_base
 ---> 575d247446ae
Step 5/27 : FROM quay.io/che-incubator/che-dashboard-next:${CHE_DASHBOARD_VERSION} as che_dashboard_next_base
 ---> 0b6b2cddb382
Step 6/27 : FROM quay.io/eclipse/che-workspace-loader:${CHE_WORKSPACE_LOADER_VERSION} as che_workspace_loader_base
 ---> fb5890b46f7d
Step 7/27 : FROM registry.centos.org/centos/centos:latest
 ---> f142359e2130
Step 8/27 : RUN localedef -i en_US -f UTF-8 C.UTF-8
 ---> Using cache
 ---> 65dd252c9e50
Step 9/27 : ENV LANG="C.UTF-8"     JAVA_HOME=/usr/lib/jvm/java-11-openjdk     PATH=${PATH}:${JAVA_HOME}/bin     CHE_HOME=/home/user/che     CHE_IN_CONTAINER=true
 ---> Using cache
 ---> 479757303383
Step 10/27 : RUN yum -y update &&     yum -y install openssl java-11-openjdk-devel sudo skopeo &&     yum clean all &&     echo "%root ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers &&     sed -i 's/Defaults    requiretty/#Defaults    requiretty/g' /etc/sudoers &&     rm -rf /tmp/* /var/cache/yum
 ---> Using cache
 ---> 8b21992f66e9
Step 11/27 : EXPOSE 8000 8080
 ---> Using cache
 ---> ffbfaa2a1757
Step 12/27 : RUN mkdir /logs /data &&     chmod 0777 /logs /data
 ---> Using cache
 ---> 645f694a0f47
Step 13/27 : COPY pcp.repo /etc/yum.repos.d/pcp.repo
 ---> Using cache
 ---> ba9dfd0c3d6a
Step 14/27 : RUN yum install -y pcp patch && yum clean all
 ---> Using cache
 ---> a413a8c06ba8
Step 15/27 : COPY ./run-pmcd.sh /run-pmcd.sh
 ---> Using cache
 ---> 36c94bca3e43
Step 16/27 : RUN chmod a+x     /run-pmcd.sh
 ---> Using cache
 ---> cf032cffeb8e
Step 17/27 : RUN mkdir -p      /etc/pcp /var/run/pcp /var/lib/pcp /var/log/pcp
 ---> Using cache
 ---> a38df1483c44
Step 18/27 : RUN chmod -R 0777 /etc/pcp /var/run/pcp /var/lib/pcp /var/log/pcp
 ---> Using cache
 ---> c2fe23f7c591
Step 19/27 : EXPOSE 44321
 ---> Using cache
 ---> f71527eef57b
Step 20/27 : COPY entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> 18ec58a670be
Step 21/27 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Using cache
 ---> 6931a9ea8974
Step 22/27 : COPY --from=che_workspace_loader_base /usr/local/apache2/htdocs/workspace-loader/ /home/user/eclipse-che/tomcat/webapps/workspace-loader
 ---> Using cache
 ---> 28f72b61b047
Step 23/27 : COPY --from=che_dashboard_base /usr/local/apache2/htdocs/dashboard /home/user/eclipse-che/tomcat/webapps/dashboard
 ---> Using cache
 ---> eea9841a3e29
Step 24/27 : COPY --from=che_dashboard_next_base /usr/local/apache2/htdocs/dashboard /home/user/eclipse-che/tomcat/webapps/dashboard/next
 ---> Using cache
 ---> edc84822659c
Step 25/27 : ADD eclipse-che /home/user/eclipse-che
 ---> 8f98cde91a89
Step 26/27 : COPY assets/branding/product.json /home/user/eclipse-che/tomcat/webapps/dashboard/assets/branding/
 ---> 8f41e3dafcc2
Step 27/27 : RUN find /home/user -type d -exec chmod 777 {} \;
 ---> Running in 4f1ecb1ef42c
Removing intermediate container 4f1ecb1ef42c
 ---> f50da0bedf91
Successfully built f50da0bedf91
Successfully tagged sleshche:rh-che-local
```
</details>

Seems I don't have access to quay.io/openshiftio/rhel-base-rh-che:latest to check Dockerfile.rhel build.

### CODE REVIEW

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
hmmm.. it fails with the failure during the image pulling:

```
Failed to pull image "quay.io/openshiftio/rhel-rhchestage-rh-che-automation:rhcheautomation-2001": [rpc error: code = Unknown desc = Error reading manifest rhcheautomation-2001 in quay.io/openshiftio/rhel-rhchestage-rh-che-automation: manifest unknown: manifest unknown, rpc error: code = Unknown desc = Error reading manifest rhcheautomation-2001 in quay.io/openshiftio/rhel-rhchestage-rh-che-automation: unauthorized: access to the requested resource is not authorized]```

@Katka92 @ScrewTSW please take a look

**Comment by ibuziuk** [comment]
[test]

File: dockerfiles/che-fabric8/Dockerfile [hunk 1]
```diff
@@ -4,9 +4,11 @@
 # which accompanies this distribution, and is available at
 # http://www.eclipse.org/legal/epl-v10.html
 ARG CHE_DASHBOARD_VERSION=next
+ARG CHE_DASHBOARD_NEXT_VERSION=next
 ARG CHE_WORKSPACE_LOADER_VERSION=next
 
 FROM quay.io/eclipse/che-dashboard:${CHE_DASHBOARD_VERSION} as che_dashboard_base
+FROM quay.io/che-incubator/che-dashboard-next:${CHE_DASHBOARD_VERSION} as che_dashboard_next_base
```
Comment (ibuziuk) [inline dockerfiles/che-fabric8/Dockerfile:11]: @sleshchenko should be `CHE_DASHBOARD_NEXT_VERSION`

File: dockerfiles/che-fabric8/Dockerfile.rhel [hunk 2]
```diff
@@ -6,9 +6,11 @@
 
 # RHEL parent dockerfile: https://github.com/rhdt/EL-Dockerfiles/blob/master/base/rh-che/Dockerfile
 ARG CHE_DASHBOARD_VERSION=next
+ARG CHE_DASHBOARD_NEXT_VERSION=next
 ARG CHE_WORKSPACE_LOADER_VERSION=next
 
 FROM quay.io/eclipse/che-dashboard:${CHE_DASHBOARD_VERSION} as che_dashboard_base
+FROM quay.io/che-incubator/che-dashboard-next:${CHE_DASHBOARD_VERSION} as che_dashboard_next_base
```
Comment (ibuziuk) [inline dockerfiles/che-fabric8/Dockerfile.rhel:13]: @sleshchenko should be CHE_DASHBOARD_NEXT_VERSION

File: dockerfiles/che-fabric8/Dockerfile (hunk 1 above)
Comment (sleshchenko) [reply dockerfiles/che-fabric8/Dockerfile]: good catch! Fixed.

**Comment by ibuziuk** [comment]
[test]

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**